src/components/PopupWithForm.js

```diff
@@ -20,6 +21,14 @@ export default class PopupWithForm extends Popup {
     super.close();
   }
 
+  renderLoading() {
+    this._submitButton.textContent = "Saving...";
+  }
+
+  endLoading() {
+    this._submitButton.textContent = this._submitButtonText;
+  }
+
```

Comment:
Needs correction: please remove line 35, you should close the popup only after the successful api response in `then` block in index.js

Reply:
Oh thank you for catching this!